### PR TITLE
Categorize errors in JDBC connector

### PR DESCRIPTION
**Pull request by cberner**

### CODE REVIEW

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ */
+package com.facebook.presto.plugin.jdbc;
+
+import com.facebook.presto.spi.ErrorCode;
+import com.facebook.presto.spi.ErrorCodeSupplier;
+
+public enum JdbcErrorCode
+        implements ErrorCodeSupplier
+{
+    // Connectors can use error codes starting at EXTERNAL
```
Comment (electrum) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java]: Remove this comment

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * A requested data conversion is not supported.
```
Comment (electrum) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java]: This doesn't make sense. If that is the what the error is supposed to mean, then please name it as such. And remove the comment.

Comment (electrum) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java]: A `SQLException` while writing could mean anything. So I think the error code is correct, but the comment is not.

However, there might be specific SQLState codes that could be given a more specific error code.

**Comment by electrum** [comment]
There are lots of other places in `JdbcRecordSink` that need to be categorized.

File: presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java (hunk 2 above)
Comment (electrum) [inline presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcErrorCode.java]: I'm thinking this should just be `JDBC_ERROR` as a more generic error code. This matches what we do in Raptor.

**Comment by cberner** [comment]
Updated

**Comment by electrum** [comment]
Much better, thanks!

**Pull request merged**